docs/getting-started/auth.adoc

```diff
@@ -1,15 +1,169 @@
 
-Mobile authentication SDK based on Keycloak
+Mobile authentication SDK based on link:http://www.keycloak.org/[Keycloak] using link:http://openid.net/connect/[OpenID Connect].
+
+Provides authentication features like access control and two factor authentication through Keycloak.
 
 == Adding dependency
 
 Add dependency to your application module
 ----
 dependencies {
-    implementation group: 'org.aerogear.android', name: 'auth'
+    implementation 'org.aerogear:auth:5.0.0-SNAPSHOT'
 }
 ----
 
 == Usage
 
+To use the Auth SDK you'll first need to:
+
+* Import the Core module. See link:./core.adoc[the Core documentation].
+* Specify the schema of the redirect url for your Android project in `android.defaultConfig.manifestPlaceholders` of the apps `build.gradle` file.
+It should match the schema specified in the `Valid Redirect URL` section for the client in Keycloak.
+It is recommended to use the package name of the Android app as the schema of the redirect url to avoid conflicts.
+
+=== Initializing the SDK
+
+`AuthService` can be retrieved using the `MobileCore#getInstance` method.
+
+[source,java]
+----
+AuthService authService = mobileCore.getInstance(AuthService.class);
+----
+
+Any subsequent `MobileCore#getInstance` method will return the same instance of `AuthService`.
+
+Before the `AuthService` can be used used `AuthService#init` must be invoked once in an app.
+
+If `AuthService#init` is not invoked then an `IllegalStateException` will be thrown when using the
+service.
+
+[source,java]
+----
+AuthServiceConfiguration authServiceConfig = new AuthServiceConfiguration.AuthConfigurationBuilder()
+    .withRedirectUri("org.aerogear.mobile.example:/callback")
+    .build();
+
+// Only invoke this once, every subsequent retrieval of the AuthService will retrieve the same, already initialized, instance.
+authService.init(context, authServiceConfig);
+----
+
+=== Retrieving the current user
+
+To retrieve the current authenticated user the `AuthService#currentUser` method can be invoked. This will be `null` if there is
+no user authenticated. So it can be used to check whether to start the authentication flow or not.
+
+[source,java]
+----
+// authService already initialized.
+UserPrincipal currentUser = authService.currentUser();
+
+if (currentUser) {
+    // User is authenticated, get the users name
+    String userName = currentUser.getName();
+    // Get the users email address
+    String userEmail = currentUser.getEmail();
+    // Get the access token of the authenticated user
+    String accessToken = currentUser.getAccessToken();
+} else {
+    // User is not authenticates, start authentication flow
+}
+----
+
+=== Authenticating
+
+To start the authentication invoke the `AuthService#login` method.
+
+[source,java]
+----
+// authService already initialized.
+AuthService authService = mobileCore.getInstance(AuthService.class);
+
+static int LOGIN_RESULT_CODE = 1;
+Activity myActivity = this.getActivity();
```

Comment:
This line doesn't make sense to me. I'm adding this code to my activity, so I use `this`.
Does that mean I'm adding it in the wrong class?